pkgs/applications/networking/browsers/firefox/common.nix

```diff
@@ -164,6 +166,20 @@ buildStdenv.mkDerivation ({
 
   inherit src unpackPhase meta;
 
+  # add profiling and a second configure phase if we build with pgo
+  phases = lib.optionals pgoSupport [
+    "unpackPhase"
+    "patchPhase"
+    "configurePhase"
+    "buildPhase"
+    "profilingPhase"
+    "configurePhase"
+    "buildPhase"
```

Comment:
Does PGO really need another full configure and build phase? Shouldn't it only need to re-link the binary?

Reply:
Ah, yes it does. PGO applies to the compiler.